backend/amd64/emit.mlp

```diff
@@ -1283,23 +1293,20 @@ let make_stack_loc ~offset i (r : Reg.t) =
 
 let emit_probe_handler_wrapper p =
   let wrap_label = probe_handler_wrapper_name p.probe_label in
-  let probe_name, handler_code_sym =
-    match p.probe_insn.desc with
-    | Lop (Iprobe { name; handler_code_sym; }) -> name, handler_code_sym
-    | _ -> assert false
-  in
+  let handler_code_sym = Option.get p.probe_handler_code_sym in
   (* Restore stack frame info as it was at the probe site, so we can easily
      refer to slots in the corresponding frame.  (As per the comment above,
      recall that the wrapper does however have its own frame.) *)
   frame_required := true;
-  stack_offset := p.stack_offset;
+  stack_offset := p.probe_stack_offset;
   for i = 0 to Proc.num_register_classes - 1 do
-    num_stack_slots.(i) <- p.num_stack_slots.(i);
+    num_stack_slots.(i) <- p.probe_num_stack_slots.(i);
   done;
   (* Account for the return address that is now pushed on the stack. *)
   stack_offset := !stack_offset + 8;
   (* Emit function entry code *)
-  D.comment (Printf.sprintf "probe %s %s" probe_name handler_code_sym);
+  D.comment (Printf.sprintf "probe %s %s" p.probe_name
+               handler_code_sym);
```

Comment:
(Does not look particularly long on one line.)

Reply:
fixed.